Main.elm

```diff
@@ -185,25 +221,27 @@ padView model =
                 "pad"
         ]
         [ controlBar model
-        , Html.div
-            [ Html.Attributes.class <|
-                case ( model.stored, model.reveal ) of
-                    ( True, True ) ->
-                        "stored reveal"
-
-                    ( True, False ) ->
-                        "stored"
-
-                    ( False, True ) ->
-                        "reveal"
-
-                    ( _, _ ) ->
-                        ""
-            , Html.Attributes.contenteditable True
-            , Html.Attributes.property "innerHTML" (Json.Encode.string model.content)
-            ]
-            []
+        , contentEditable model
+        ]
+
+
+innerHtmlDecoder =
+    Json.Decode.at [ "target", "innerHTML" ] Json.Decode.string
+
+
+contentEditable : Model -> Html.Html Msg
+contentEditable model =
+    Html.div
+        [ Html.Attributes.class <|
+            if model.reveal then
+                "reveal"
+            else
+                ""
```

Comment:
Yes, clicking on the button blurs the `contentEditable`, so we don't care about the caret position anymore.

By the way, redrawing the `contentEditable` isn't the issue it seems, it's only when changing the `innerHTML` that we would loose the caret position (which would jump back to the beginning).

The fix is thus to not change the `innerHTML` unless it's loaded from the store (and thus erases anything already there, making it ok to loose the caret position). The `innerHTML` is thus `loadedContent`, which will be out of sync with `content` as soon as the user starts typing (but we don't really care, as it's the `content` that's used to save the latest state of the content, and then stored).